src/main/drivers/flash_w25n01g.c

```diff
@@ -38,16 +38,19 @@
 // Device size parameters
 #define W25N01G_PAGE_SIZE         2048
 #define W25N01G_PAGES_PER_BLOCK   64
-#define W25N01G_BLOCKS_PER_DIE 1024
+#define W25N01G_BLOCKS_PER_DIE    1024
+
+#define W25N02K_PAGE_SIZE         2048
+#define W25N02K_PAGES_PER_BLOCK   64
+#define W25N02K_BLOCKS_PER_DIE    2048
 
 // BB replacement area
 #define W25N01G_BB_MARKER_BLOCKS           1
 #define W25N01G_BB_REPLACEMENT_BLOCKS      20
 #define W25N01G_BB_MANAGEMENT_BLOCKS       (W25N01G_BB_REPLACEMENT_BLOCKS + W25N01G_BB_MARKER_BLOCKS)
 // blocks are zero-based index
-#define W25N01G_BB_REPLACEMENT_START_BLOCK (W25N01G_BLOCKS_PER_DIE - W25N01G_BB_REPLACEMENT_BLOCKS)
-#define W25N01G_BB_MANAGEMENT_START_BLOCK  (W25N01G_BLOCKS_PER_DIE - W25N01G_BB_MANAGEMENT_BLOCKS)
-#define W25N01G_BB_MARKER_BLOCK            (W25N01G_BB_REPLACEMENT_START_BLOCK - W25N01G_BB_MARKER_BLOCKS)
+#define W25N01G_BB_REPLACEMENT_START_BLOCK (fdevice->geometry.sectors - W25N01G_BB_REPLACEMENT_BLOCKS)
+#define W25N01G_BB_MANAGEMENT_START_BLOCK  (fdevice->geometry.sectors - W25N01G_BB_MANAGEMENT_BLOCKS)
```

Comment:
This would be easier to read imho
```suggestion
#define W25N01G_BB_REPLACEMENT_START_BLOCK (W25N01G_BLOCKS_PER_DIE - W25N01G_BB_REPLACEMENT_BLOCKS)
#define W25N01G_BB_MANAGEMENT_START_BLOCK  (W25N01G_BLOCKS_PER_DIE - W25N01G_BB_MANAGEMENT_BLOCKS)
```

Also `W25N01G_BB_REPLACEMENT_START_BLOCK` seems not be used so we could do

```suggestion
#define W25N01G_BB_MANAGEMENT_START_BLOCK  (W25N01G_BLOCKS_PER_DIE - W25N01G_BB_MANAGEMENT_BLOCKS)
```

Reply:
This is an intended change as the 01G part has 1024 blocks per die, and the 02K has 2048. I'll rename the `W25N01G_BB_REPLACEMENT_START_BLOCK` and `W25N01G_BB_MANAGEMENT_START_BLOCK` macros to avoid confusion.